ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/RegisterTopicCommand.java

```diff
@@ -101,11 +102,10 @@ public DdlCommandResult run(final MetaStore metaStore) {
     if (metaStore.getTopic(topicName) != null) {
       // Check IF NOT EXIST is set, if set, do not create topic if one exists.
       if (notExists) {
-        return new DdlCommandResult(true,
-                                    "Topic is not registered because it already registered"
-                                          + ".");
+        return new DdlCommandResult(true, "Topic already registered.");
       } else {
-        throw new KsqlException("Topic already registered: " + topicName);
+        throw new KsqlException(String.format("Failed to create because %s with name '%s' "
```

Comment:
Nit: I'd prefer a noun after `create`, at the moment it feels a bit fragmentary. It's clearer than before but whilst we've got the hood up let's nail it :) 

Something like: 

* `Failed to create stream because stream with name x already exists`
* `Failed to create table because stream with name x already exists`

etc

Reply:
I'd like to do that too, but I would need to add  new parameter to the `RegisterTopicCommand.run() `just to specify the topic type it is trying to create in order to use it in the error message. I don't like that approach just for a custom message.

However, I think I can play with a message thrown by the `RegisterTopicCommand.run()` call and add a new one from the `CreateStreamCommand` and `CreateTableCommand` and make it look better. 

What do you think about this message:

- `Cannot create table 't1': Stream with name 't1' already exists`
- `Cannot create stream 't1': Stream with name 't1' already exists`

If that works, then I can do it in the code like this:

```
run(final MetaStore metaStore) {
  throw KsqlException("%s with name '%s' already exists", sourceType, topicName);
}

...

try {
  run(metastore);
} catch (KsqlException e) {
  throw new KsqlException("Cannot create table '%s': " + e.getMessage(), tableName);
}
```

Reply:
Yes that works for me. I'd want consistent capitalisation of `stream`/`Stream` and `table`/`Table` though ;)
Can you share a sample of the actual output messages once you've tested this?

Reply:
> I'd like to do that too, but I would need to add new parameter to the `RegisterTopicCommand.run() `just to specify the topic type it is trying to create in order to use it in the error message. I don't like that approach just for a custom message.

FYI, a topic can be registered outside of a stream or table being created. So I'm not sure that would work anyway.

Reply:
I found other issues, though. If I register a topic from ksql using a topic name that I registered from the Confluent UI, then the command works without complaining the topic already existed. Same for streams, if I create a stream from an already registered topic, then it works. 

I will need to fix more places to avoid those situations instead of just fixing the error message. Also, there is a different message when creating a table with a name from a stream:
`Cannot add the new data source. Another data source with the same name already exists: KsqlStream name:STREAM1`

Reply:
Not sure I follow.

By Confluent UI I take it you mean the C3 web UI?   By `Ksql` do you mean the KSQL CLI?

As I've said else where the whole `REGISTER TOPIC` stuff isn't really supported / documented and has bugs and we should rip it out for this very reason.

I'd focus this PR on just fixing issues with CS, CSAS, CT and CTAS statements. 

Feel free to raise follow up PRs to fix other issues or, my preference, rip out `REGISTER TOPIC` functionality,

Reply:
Yes, by Confluent UI is by using the C3 web UI. I chat with @hjafarpour yesterday btw, and he said there might be future plans to keep the `register topic`. I will leave the generic error message there.